include/event_sources/timer.h

```diff
@@ -28,12 +28,57 @@ enum class TimerType
   INTERVAL,
 };
 
+class BaseTimer : public virtual microloop::EventSource
```

Comment:
Is a base class _really_ neccessary? Will it ever have more than one concrete implementation?

Reply:
The `BaseTimer` is needed so I don't need `TimerController` to be a template. It does not depend on the callback type so it would just be overhead to make it a template.